terraform/canary/amis.tf

```diff
@@ -52,21 +52,19 @@ variable "amis" {
   default = {
     canary_windows = {
       os_family          = "windows"
-      ami_search_pattern = "Windows_Server-2019-English-Full-Base-*"
+      ami_search_pattern = "Windows_Server-2022-English-Full-Base*"
       ami_owner          = "amazon"
-      ami_id             = "ami-01f14dc60171d8d7b"
       ami_product_code   = []
       family             = "windows"
-      arch               = "amd64"
+      arch               = "x86_64"
     }
     canary_linux = {
       os_family          = "amazon_linux"
-      ami_search_pattern = "amzn2-ami-hvm-2.0.????????.?-x86_64-gp2"
+      ami_search_pattern = "amzn2-ami-kernel*"
```

Comment:
Do we need to choose a kernel versions here, or does this pattern only catch the kernel 5.x AMIs?

Reply:
It catches the latest kernel version. Currently it is 5.x; but it should ideally pick 6.x if `amzn2-ami-kernel.6.x` is out

Reply:
Does it catch the latest kernel version, or the latest AMI?  If there are both `kernel.5.x` and `kernel.6.x` AMIs available would it consistently get the 6.x version or could it get 5.x one time and 6.x another depending on which was published most recently?

Do we still support the 4.x AL2 releases?  I believe those are the ones with no `kernel` in the AMI name.  Should we have two entries, one for `hvm-*` and one for `kernel.5*`?  We can always add another for `kernel.6*` if it becomes necessary, though I expect that won't happen on AL2.

Reply:
>Does it catch the latest kernel version, or the latest AMI? If there are both kernel.5.x and kernel.6.x AMIs available would it consistently get the 6.x version or could it get 5.x one time and 6.x another depending on which was published most recently?

That is a good point, I did not think about that edge case. Even-though idk how to exactly test out this behavior ; the terraform documentation mentions that it pulls the` recent results`. So it will probably pull the `kernel.5.1.x` if it is updated after `kernel.6.x`. I think it is better to update the search patter to `amzn2-ami-kernel.5*` and update to `amzn2-ami-kernel.6*` when its out.

>Do we still support the 4.x AL2 releases? I believe those are the ones with no kernel in the AMI name. Should we have two entries, one for hvm-* and one for kernel.5*? We can always add another for kernel.6* if it becomes necessary, though I expect that won't happen on AL2.

Should we ? The initial plan was just to test out the latest version of AL2. The [release notes ](https://aws.amazon.com/about-aws/whats-new/2021/11/amazon-linux-2-ami-kernel-5-10/)for kernal 5 mention various security improvements. So IMO we should suggest users to upgrade.

Reply:
Updated the search pattern in [ef1a2a0](https://github.com/aws-observability/aws-otel-test-framework/pull/856/commits/ef1a2a0a0d6c13acfb8d84f54fbc0bfa62f41e4c)